### PR TITLE
doc: add example of deploying api server to Kubernetes

### PR DESCRIPTION
**Pull request by uzuku**
## Modification

Doc: add example of deploying api server to Kubernetes.

### CODE REVIEW

File: docs/en/serving/api_server.md [hunk 1]
```diff
@@ -37,6 +37,105 @@ docker run --runtime nvidia --gpus all \
...
+        hostPath:
+          path: /root/.cache/huggingface
+          type: DirectoryOrCreate
+      - emptyDir:
+          medium: Memory
+        name: dshm
+EOF
+```
+
+In the example above the model data is placed on the local disk of the node (hostPath). Consider replacing it with high-availability shared storage if multiple replicas are desired, and the storage can be mounted into container using [PersistentVolume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
```
Comment (flyhighzy) [inline docs/en/serving/api_server.md:137]: 还有提醒下需要填写huggingface token，或者提出来在开始赋值一个shell变量比较明显

**Comment by lvhan028** [comment]
Basically, I agree with @zhyncs
but how about making `k8s` folder in the root directory?
The config content mentioned can be put in a yaml under that folder.
And the `api_server` user guide can refer to this yaml as an example

**Comment by uzuku** [comment]
@lvhan028 @zhyncs I've put the related yaml files under the k8s folder in the root directory.

**Comment by lvhan028** [comment]
pls fix the linting error
```
pip install pre-commit
cd the/root/path/of/lmdeploy
pre-commit install .
pre-commit run --all-files
```

**Comment by uzuku** [comment]
> pls fix the linting error

@lvhan028 fixed

**Comment by flyhighzy** [comment]
> Basically, I agree with @zhyncs but how about making `k8s` folder in the root directory? The config content mentioned can be put in a yaml under that folder. And the `api_server` user guide can refer to this yaml as an example

The `api_server` folder under k8s may mislead to kubernetes' own api server, how about rename the yaml's fold to `k8s/example`?

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**